hazelcast/src/main/java/com/hazelcast/sql/impl/exec/fetch/Fetch.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.exec.fetch;
+
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.ExpressionEvalContext;
+import com.hazelcast.sql.impl.row.EmptyRowBatch;
+import com.hazelcast.sql.impl.row.ListRowBatch;
+import com.hazelcast.sql.impl.row.Row;
+import com.hazelcast.sql.impl.row.RowBatch;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * LIMIT/OFFSET core support.
+ */
+public class Fetch {
+
+    private final Expression<?> fetch;
```

Comment:
The `fetch` and `offset` expressions must not have references to the row. Therefore, they could be evaluated before the query execution. This change will simplify the execution code but would require some additional changes to the pre-execution phase (e.g. apply parameters to the node, to construct the new node with resolved values). 
We may plan this improvement in a separate issue.

Reply:
Now I see that we do it in the validator. Resolving.